surface/src/on_nurbs/fitting_curve_2d_apdm.cpp

```diff
@@ -1046,7 +1046,7 @@ FittingCurve2dAPDM::findClosestElementMidPoint (const ON_NurbsCurve &nurbs, cons
     double &xi1 = elements[i + 1];
     double dxi = xi1 - xi0;
 
-    for (unsigned j = 0; j < nurbs.Order (); j++)
+    for (unsigned j = 0; j < static_cast<unsigned int>(nurbs.Order ()); j++)
```

Comment:
The casts are all over the place. (to unsigned int, size_t). Maybe only one is required and not all (for `nurbs.Order()`) variants